### PR TITLE
More info for log messages

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
Added file name, line number, thread id (NOT platform independent), and current time to log messages.

### CODE REVIEW

File: driver/driver.d [hunk 1]
```diff
@@ -845,7 +845,6 @@ export SQLRETURN SQLTablesW(
 
             latestOdbcResult = result;
             return SQL_SUCCESS;
-            return SQL_SUCCESS;
```
Comment (nezihyigitbasi) [inline driver/driver.d:848]: Interesting that the compiler doesn't complain about the last return.

File: driver/util.d [hunk 2]
```diff
@@ -21,6 +21,11 @@ import std.file : append, write, readText, remove, exists, FileException;
 import std.traits : isNumeric, isIntegral, isSomeString, isSomeChar, Unqual;
 import core.stdc.stdlib : abort;
 import core.exception : Exception;
+import core.time : dur, Duration;
+import std.net.curl : HTTP, CurlException;
+import std.string: format;
+import core.sys.windows.windows;//TODO: need platform independence
```
Comment (nezihyigitbasi) [inline driver/util.d]: couldn't find a platform independent way of getting the thread ID.

File: driver/driver.d (hunk 1 above)
Comment (markisaa) [inline driver/driver.d:848]: Good catch. Yes - thanks for the reminder. Warnings are currently not enabled. I will make a pass at that and enable warnings and warnings as errors in a near-future pull request.

File: driver/util.d (hunk 2 above)
Comment (markisaa) [inline driver/util.d]: version (Windows) {
  import core.sys.windows.windows;
}

Looks like you did this for `GetCurrentThreadId`. You should not do this for two reasons:
1. This driver is single threaded
2. This is not portable. See `std.concurrency` for a better option. That said, let's not log this until if/when we actually support concurrency in the driver.

Comment (nezihyigitbasi) [inline driver/util.d]: Alright getting rid of that.

**Comment by markisaa** [comment]
Merged.

Made some edits before merging, might be worth checking out: https://github.com/prestodb/presto-odbc/commit/b219e7028d5e8eece87e92979b12fa9a9cbcbb96

Summary:
1. Only using `__FILE__` and `__LINE__` in `buildDebugMessage` meant that it always said `util.d:40` or so, so I changed it to forward those values from calls to `logMessage` and `logCriticalMessage`
2. Added a comment to explain why the `version (unittest)` shortcut was in there.
3. Added padding to the call to `format` so that the log is still pretty nicely arranged/human readable.
4. Took the opportunity to remove some deprecated code for printing C-strings and `static assert` in that case instead.

**Pull request closed without merging**